jdbc/src/main/java/com/interface21/transaction/support/TransactionSynchronizationManager.java

```diff
@@ -8,16 +10,28 @@ public abstract class TransactionSynchronizationManager {
 
     private static final ThreadLocal<Map<DataSource, Connection>> resources = new ThreadLocal<>();
 
+    static {
+        resources.set(new HashMap<>());
+    }
+
     private TransactionSynchronizationManager() {}
 
     public static Connection getResource(DataSource key) {
-        return null;
+        return resources.get().get(key);
     }
 
     public static void bindResource(DataSource key, Connection value) {
+        if (getResource(key) != null) {
+            throw new DataAccessException("Transaction already started for this datasource");
+        }
```

Comment:
```suggestion
        if (resources.get().containsKey(key)) {
            throw new DataAccessException("Transaction already started for this datasource");
        }
```

여기도 이렇게 해도 될 것 같아요!!

+) 단순 질문인데 예외를 터뜨린 이유가 무엇이었나요?!
저는 예외 처리까지는 안 했어서 (어차피 덮어씌워지니까) 어떠한 관점에서 예외 상황이라고 판단했는지 궁금해요!